src/lib.rs

```diff
@@ -1530,6 +1530,21 @@ mod tests {
         assert_eq!(BFloat16::from(1.5f32).to_string(), "1.5");
     }
 
+    #[test]
+    fn test_f16() {
+        let data: Vec<f16> = vec![-1.0f32, 0.0, 1.0, 2.5]
+            .into_iter()
+            .map(|x| f16::from_f32(x))
+            .collect();
+        let tensor = <Tensor<f16>>::new(&[2, 2]).with_values(&data);
+        assert!(tensor.is_ok());
+        let tensor = tensor.unwrap();
+        assert_eq!(tensor.len(), 4);
+        for i in 0..data.len() {
+            assert_eq!(tensor[i], data[i]);
```

Comment:
You can make this more succinct by replacing

```
assert_eq!(tensor.len(), 4);
for i in 0..data.len() {
    assert_eq!(tensor[i], data[i]);
}
```

with

```
assert_eq!(&tensor[..], &data[..]);
```

Reply:
Sure :) done